Modules/Devices/src/Devices.Infrastructure/PushNotifications/IServiceCollectionExtensions.cs

```diff
@@ -21,20 +26,31 @@ public static void AddPushNotifications(this IServiceCollection services, PushNo
             case PROVIDER_DUMMY: 
                 services.AddDummyPushNotifications();
                 break;
+            case PROVIDER_PNS_CONNECTOR:
+
+                FirebaseApp.Create(new AppOptions()
+                {
+                    Credential = GoogleCredential.FromJson(options.FCM_Service_Account),
+                });
+
+                services.AddTransient<PnsConnectorFactory, PnsConnectorFactoryImpl>();
+                services.AddTransient<FirebaseCloudMessagingConnector>();
+                services.AddTransient<IPushService, DirectPushService>();
+                break;
             default:
                 throw new Exception($"Push Notification Provider {options.Provider} does not exist.");
         }
-
-        services.AddTransient<PnsConnectorFactory, PnsConnectorFactoryImpl>();
     }
 }
 
 public class PushNotificationOptions
 {
     [Required]
     [RegularExpression(
-        $"{IServiceCollectionExtensions.PROVIDER_AZURE_NOTIFICATION_HUB}|{IServiceCollectionExtensions.PROVIDER_DUMMY}")]
+        $"{IServiceCollectionExtensions.PROVIDER_AZURE_NOTIFICATION_HUB}|{IServiceCollectionExtensions.PROVIDER_PNS_CONNECTOR}|{IServiceCollectionExtensions.PROVIDER_DUMMY}")]
     public string Provider { get; set; } = IServiceCollectionExtensions.PROVIDER_AZURE_NOTIFICATION_HUB;
 
     public AzureNotificationHub.IServiceCollectionExtensions.AzureNotificationHubPushNotificationsOptions AzureNotificationHub { get; set; }
+
+    public string FCM_Service_Account { get; set; }
```

Comment:
You should do something similar to the AzureNotificationHub options.

```c#
public DirectPnsCommunicationOptions DirectPnsCommunication { get; set; }
```

```c#
public class DirectPnsCommunicationOptions
{
    public FcmOptions Fcm { get; set; }

    public class FcmOptions
    {
        public string? ServiceAccountJson { get; set; } = string.Empty;
    }
}
```

And when you use it, you should check if ServiceAccountJson is null, and if so, you should use the default credentials:

```c#
FirebaseApp.Create(new AppOptions()
{
    Credential = GoogleCredential.GetApplicationDefault(),
});
```